invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -178,10 +185,20 @@ def update(self, pid, record, url=None, **kwargs):
         :param record: the record metadata for the DOI.
         :returns: `True` if is updated successfully.
         """
+        hide = False
+        if isinstance(record, ChainObject):
+            if record._child["access"]["record"] == "restricted":
+                hide = True
+        elif record["access"]["record"] == "restricted":
+            hide = True
+
         try:
-            # Set metadata
-            doc = self.serializer.dump_obj(record)
-            self.client.api.update_doi(metadata=doc, doi=pid.pid_value, url=url)
+            if hide:
+                self.client.api.hide_doi(doi=pid.pid_value)
+            else:
```

Comment:
Another way could be to delegate the state to DataCite: we could first read/get from DataCite, check if it is hidden or not, and therefore add the `event: publish` or not. We are in a celery task so it could work.
However, we need to take into account that DataCite might not respond, and what to do in that case.
At the same time, I think that it would make sense to have an `update` method that receive the old and the new version of the record, so the implementation could use the diff to make choices. It feels like a much bigger change.

Reply:
which concerns do you have with having always the `event: publish` there?

Reply:
in my opinion it would complicate the whole process if it has first to be checked on DataCite if the record is hidden. The state management will be more complex and we need at least one request to get the information. Yes one request is not that much but scaled on multiple InvenioRDM instances with multiple records DataCite has to handle a lot more requests.